ethereum/spec/src/test/java/tech/pegasys/teku/spec/ForkManifestTest.java

```diff
@@ -0,0 +1,90 @@
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.datastructures.state.Fork;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.constants.SpecConstants;
+import tech.pegasys.teku.ssz.SSZTypes.Bytes4;
+
+class ForkManifestTest {
+  private final Bytes4 genesisForkVersion = Bytes4.fromHexString("0x00000000");
+  private final Bytes4 secondForkVersion = Bytes4.fromHexString("0x11111111");
+  private final Bytes4 thirdForkVersion = Bytes4.fromHexString("0x22222222");
+  private final Fork genesisFork = new Fork(genesisForkVersion, genesisForkVersion, UInt64.ZERO);
+  private final Fork secondFork =
+      new Fork(genesisForkVersion, secondForkVersion, UInt64.valueOf(100L));
+  private final Fork thirdFork =
+      new Fork(secondForkVersion, thirdForkVersion, UInt64.valueOf(200L));
+  private final SpecConstants constants = mock(SpecConstants.class);
+  private ForkManifest forkManifest;
+
+  @BeforeEach
+  public void setup() {
+    when(constants.getGenesisForkVersion()).thenReturn(genesisForkVersion);
+    when(constants.getGenesisEpoch()).thenReturn(0L);
+    forkManifest = ForkManifest.create(List.of(genesisFork, secondFork, thirdFork));
+  }
+
+  @Test
+  void shouldNotAllowAppendingEarlierFork() {
+    assertThrows(
+        IllegalArgumentException.class, () -> ForkManifest.create(List.of(thirdFork, secondFork)));
+  }
+
+  @Test
+  void shouldNotAllowPreviousVersionMismatch() {
+    final Fork fork = new Fork(secondForkVersion, thirdForkVersion, UInt64.valueOf(1000L));
+    assertThrows(
+        IllegalArgumentException.class, () -> ForkManifest.create(List.of(genesisFork, fork)));
+  }
+
+  @Test
+  void shouldNotAllowCurrentAndPreviousVersionToBeTheSame() {
+    final Fork fork = new Fork(genesisForkVersion, genesisForkVersion, UInt64.valueOf(1000L));
+    assertThrows(
+        IllegalArgumentException.class, () -> ForkManifest.create(List.of(genesisFork, fork)));
+  }
+
+  @Test
+  void shouldGetEpochForkAfterManifestCreation() {
+    assertThat(forkManifest.get(UInt64.ZERO)).isEqualTo(genesisFork);
+    assertThat(forkManifest.get(UInt64.valueOf(99L))).isEqualTo(genesisFork);
+  }
+
+  @Test
+  void shouldGetGenesisForkFromManifest() {
+    assertThat(forkManifest.getGenesisFork()).isEqualTo(genesisFork);
+  }
+
+  @Test
+  void shouldGetMiddleFork() {
+    assertThat(forkManifest.get(UInt64.valueOf(100L))).isEqualTo(secondFork);
+    assertThat(forkManifest.get(UInt64.valueOf(199L))).isEqualTo(secondFork);
+  }
+
+  @Test
+  void shouldGetLastForkFromSchedule() {
+    assertThat(forkManifest.get(UInt64.valueOf(200L))).isEqualTo(thirdFork);
+    assertThat(forkManifest.get(UInt64.MAX_VALUE)).isEqualTo(thirdFork);
+  }
```

Comment:
We should add a couple extra tests here:
* Try to create a `ForkManifest` by passing in an empty list
* Try to create a `ForkManifest` with only the 2nd and 3rd fork (no genesis fork)